minorminer/utils/raster_embedding.py

```diff
@@ -0,0 +1,437 @@
+# Copyright 2024 D-Wave
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import networkx as nx
+import dwave_networkx as dnx
+import numpy as np
+import warnings
+import matplotlib.pyplot as plt
+from minorminer.subgraph import find_subgraph
+
+def visualize_embeddings(H, topology=None, title=None):
+    """
+    Visualizes the embeddings produced using dwave_networkx's layout 
+    according to the specified topology.
+
+    Parameters:
+    -----------
+    H : networkx.Graph
+        Input graph to be visualized on the topology.
+    topology : str, optional
+        The type of topology for visualization. 
+    title : str, optional
+        Title of the plot. Defaults to None.
+
+    """
+
+    if topology == 'chimera':
+        graph_size=(1, 1, 4)
+        # Set up the figure and axis
+        fig, ax = plt.subplots(figsize=(8, 6))
+        
+        # Create the Chimera topology graph (adjust size as needed)
+        G = dnx.chimera_graph(*graph_size)  
+        
+        # Draw the base Chimera unit cell
+        dnx.draw_chimera(G, node_color='r', ax=ax, label='Unit Cell')
+
+        # Draw the input graph on top of the unit cell
+        dnx.draw_chimera(
+            H, 
+            node_color='b', 
+            node_shape='*', 
+            style='dashed', 
+            edge_color='b', 
+            width=3, 
+            ax=ax, 
+            label='Input Graph'
+        )
+
+        # Add title if provided
+        if title:
+            ax.set_title(title)
+
+        # Display the legend
+        ax.legend(loc='best')
+        plt.show()
+    elif topology == 'pegasus':
+        fig, ax = plt.subplots(figsize=(8,6))
+
+        G = dnx.pegasus_graph(2)
+
+        dnx.draw_pegasus(G, with_labels=True, crosses=True, node_color="Yellow", ax=ax)
```

Comment:
why the special case for pegasus (`crosses=True`, `node_color="Yelow"`).
if it's not necessary to visualize the three topologies with different drawing parameters, u can do something like
```
elif topology == "pegasus":
    draw = dnx.draw_pegasus
    ...
...
draw(G, params=...)
```